src/coreclr/src/jit/codegencommon.cpp

```diff
@@ -6308,71 +6308,131 @@ void CodeGen::genZeroInitFrame(int untrLclHi, int untrLclLo, regNumber initReg,
         noway_assert(uCntBytes == 0);
 
 #elif defined(TARGET_XARCH)
-        /*
-            Generate the following code:
-
-                lea     edi, [ebp/esp-OFFS]
-                mov     ecx, <size>
-                xor     eax, eax
-                rep     stosd
-         */
+        noway_assert(compiler->getSIMDSupportLevel() >= SIMD_SSE2_Supported);
+        // As we output multiple instructions for SIMD zeroing, we want to balance code size, with throughput
+        // so cap max size at 16 * MAX SIMD length
+        int initMaxSIMDSize = 16 * XMM_REGSIZE_BYTES;
+        if ((untrLclHi - untrLclLo) <= initMaxSIMDSize)
```

Comment:
There may be locals in this region that do not need to be zero-initialized. Do it add up to anything significant? We may be able to do better by skipping zero-initialization for them.

Reply:
Is handled by the else branch when `genUseBlockInit` is false where it just zeros out the GC fields (not sure of heuristics that determine it); starts just after the shown GitHub diff at `else if (genInitStkLclCnt > 0)`

Reply:
E.g. it may be worth it to build a bit-mask what needs to be zero-initialized, and then go over the bitmask to do the zero-init.

Reply:
Do have an issue open for double zeroing https://github.com/dotnet/runtime/issues/2325

Reply:
Could use use the same logic the `genInitStkLclCnt` else block uses to drop any `xmm` blocks that don't have something to zero? (in a follow up)